src/librustdoc/html/markdown.rs

```diff
@@ -1034,12 +1034,97 @@ impl MarkdownSummaryLine<'_> {
+/// due to the provided `length_limit`.
+fn render_markdown_summary_with_limit(md: &str, length_limit: Option<u16>) -> (String, bool) {
+    if md.is_empty() {
+        return (String::new(), false);
+    }
+
+    let length_limit = length_limit.unwrap_or(u16::MAX) as usize;
+
+    let mut s = String::with_capacity(md.len() * 3 / 2);
+    let mut text_length = 0;
+    let mut stopped_early = false;
+
+    'outer: for event in Parser::new_ext(md, Options::ENABLE_STRIKETHROUGH) {
+        match &event {
+            Event::Text(text) => {
+                for (i, word) in text.split(' ').enumerate() {
+                    if text_length + word.len() >= length_limit {
+                        stopped_early = true;
+                        break 'outer;
+                    }
+
+                    if i > 0 {
+                        s.push(' ');
+                        text_length += 1;
+                    }
+
+                    s.push_str(word);
+                    text_length += word.len();
+                }
+            }
+            Event::Code(code) => {
+                if text_length + code.len() >= length_limit {
+                    stopped_early = true;
+                    break;
+                }
+
+                s.push_str("<code>");
+                s.push_str(code);
+                s.push_str("</code>");
+                text_length += code.len();
+            }
+            Event::Start(tag) => match tag {
+                Tag::Emphasis => s.push_str("<em>"),
+                Tag::Strong => s.push_str("<strong>"),
+                Tag::CodeBlock(..) => break,
+                _ => {}
+            },
+            Event::End(tag) => match tag {
+                Tag::Emphasis => s.push_str("</em>"),
+                Tag::Strong => s.push_str("</strong>"),
+                Tag::Paragraph => break,
+                _ => {}
+            },
+            Event::HardBreak | Event::SoftBreak => s.push(' '),
+            _ => {}
+        }
+    }
+
+    (s, stopped_early)
+}
+
+/// Renders a shortened first paragraph of the given Markdown as a subset of Markdown,
+/// making it suitable for contexts like the search index.
+///
+/// Will shorten to 59 or 60 characters, including an ellipsis (…) if it was shortened.
+///
+/// See [`render_markdown_summary_with_limit`] for details about what is rendered and what is not.
+crate fn render_short_markdown_summary(markdown: &str) -> String {
+    let (mut s, was_shortened) = render_markdown_summary_with_limit(markdown, Some(59));
+
+    if was_shortened {
+        s.push('…');
+    }
+
+    s
+}
+
 /// Renders the first paragraph of the provided markdown as plain text.
+/// Useful for alt-text.
 ///
 /// - Headings, links, and formatting are stripped.
 /// - Inline code is rendered as-is, surrounded by backticks.
 /// - HTML and code blocks are ignored.
-pub fn plain_text_summary(md: &str) -> String {
+crate fn render_plain_text_summary(md: &str) -> String {
```

Comment:
I liked this before, is there a reason to change it? Being called `render` without being in the `render` module seems confusing.

Reply:
I guess not; I just try to have function names be actions, rather than nouns :)

Reply:
And I also wanted it be consistent with `render_markdown_summary*`.